src/main/java/org/kaazing/netx/ws/internal/io/WsMessageReader.java

```diff
@@ -188,18 +189,21 @@ public WsMessageReader(WsURLConnectionImpl connection) throws IOException {
             throw new NullPointerException(MSG_NULL_CONNECTION);
         }
 
+        int maxFrameLength = FrameUtil.calculateCapacity(false, connection.getMaxPayloadLength());
+
```

Comment:
Consider hoisting this out to reuse recalculation for each type of reader / input stream.
Do we still need `FrameUtil` ?